File: CommonUILibrary/src/main/java/com/zh/commonuilibrary/util/DensityUtils.java
package com.zh.commonuilibrary.util;

import android.content.Context;
import android.util.DisplayMetrics;
import android.util.TypedValue;

/**
 * @describe:
 * @author: Z H
 * @date: 2020/3/2 16:27
 * @pkgName: com.zh.commonuilibrary.util
 */
public class DensityUtils {

    public static int dip2px(Context context, float dpVal) {
        return dp2px(context, dpVal);
    }

    public DisplayMetrics getDisplayMetrics(Context context) {
        return context.getResources().getDisplayMetrics();
    }

    public static int dp2px(Context context, float dpVal) {
        return (int) TypedValue.applyDimension(1, dpVal, context.getResources().getDisplayMetrics());
    }

    public static int sp2px(Context context, float spVal) {
        return (int) TypedValue.applyDimension(2, spVal, context.getResources().getDisplayMetrics());
    }

    public static float px2dp(Context context, float pxVal) {
        float scale = context.getResources().getDisplayMetrics().density;
        return pxVal / scale;
    }

    public static float px2sp(Context context, float pxVal) {
        return pxVal / context.getResources().getDisplayMetrics().scaledDensity;
    }


    public static int getPhoneWidth(Context context) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        return dm.widthPixels;
    }

    public static int getPhoneHeight(Context context) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        return dm.heightPixels;
    }
}

File: CommonUILibrary/src/main/java/com/zh/commonuilibrary/util/DensityUtils.kt
package com.zh.commonuilibrary.util

import android.content.Context
import android.util.TypedValue

/**
 * @describe: 设备尺寸像素相关
 * @author: Z H
 * @date: 2020/3/2 16:27
 * @pkgName: com.zh.commonuilibrary.util
 */
fun dp2px(context: Context, dpVal: Float): Int {
    return TypedValue.applyDimension(1, dpVal, context.resources.displayMetrics).toInt()
}

fun sp2px(context: Context, spVal: Float): Int {
    return TypedValue.applyDimension(2, spVal, context.resources.displayMetrics).toInt()
}

fun px2dp(context: Context, pxVal: Float): Float {
    val scale = context.resources.displayMetrics.density
    return pxVal / scale
}

fun px2sp(context: Context, pxVal: Float): Float {
    return pxVal / context.resources.displayMetrics.scaledDensity
}

fun getPhoneWidth(context: Context): Int {
    val dm = context.resources.displayMetrics
    return dm.widthPixels
}

fun getPhoneHeight(context: Context): Int {
    val dm = context.resources.displayMetrics
    return dm.heightPixels
}